packages/common/src/compose-tunnel-agent.ts

```diff
@@ -18,3 +18,55 @@ export type ScriptInjection = {
   defer?: boolean
   async?: boolean
 }
+
+type Stringified<T> = {
+  [k in keyof T]: string
+}
+const parseScriptInjection = ({ pathRegex, defer, async, src }: Stringified<ScriptInjection>):
+ ScriptInjection | Error => {
+  try {
+    if (!src) {
+      throw new Error('missing src')
+    }
+    return {
+      ...pathRegex && { pathRegex: new RegExp(pathRegex) },
+      ...defer && { defer: defer === 'true' },
+      ...async && { async: async === 'true' },
+      src,
+    }
+  } catch (e) {
+    return e as Error
+  }
+}
+
+export const extractSectionsFromLabels = <T>(prefix: string, labels: Record<string, string>) => {
+  const re = new RegExp(`^${prefix.replace(/\./g, '\\.')}\\.(?<id>.+?)\\.(?<key>[^.]+)$`)
```

Comment:
It was originally like that, but it requires more from the consumer of `extractSectionsFromLabels` and it can create variance in the configuration in the way we define section which is not something we want, I believe that in a future iteration I'll be probably change it back to not using Regex instead of composing the expression.

Reply:
Updated the implementation to not use Regex